django/db/models/aggregates.py

```diff
@@ -211,3 +213,30 @@ def __init__(self, expression, sample=False, **extra):
 
     def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "VAR_SAMP"}
+
+
+class JSONArrayAgg(Aggregate):
+    function = "JSON_ARRAYAGG"
+    name = "JSONArrayAgg"
+    output_field = JSONField()
+    arity = 1
+
+    def as_sqlite(self, compiler, connection, **extra_context):
+        return super().as_sql(
+            compiler, connection, function="JSON_GROUP_ARRAY", **extra_context
+        )
+
+    def as_postgresql(self, compiler, connection, **extra_context):
+        if not connection.features.is_postgresql_16:
+            sql, params = super().as_sql(
+                compiler,
+                connection,
+                function="ARRAY_AGG",
+                template="TO_JSON(%(function)s(%(distinct)s%(expressions)s))",
+                **extra_context,
+            )
+        else:
+            sql, params = super().as_sql(compiler, connection, **extra_context)
+        if self.output_field.get_internal_type() == "JSONField":
+            sql = "CAST(%s AS JSONB)" % sql
```

Comment:
Pretty sure this branch will always be true. Is there any reason not to always use `TO_JSONB` and `{sql}RETURNING jsonb` [instead](https://www.postgresql.org/docs/current/functions-aggregate.html#id-1.5.8.27.5.2.4.18.1.1.1)?